### PR TITLE
[VR-10896] Populate all code version from blob

### PR DESCRIPTION
**Pull request by anandjakhaniya**
**Jira Ticket:** https://vertaai.atlassian.net/browse/VR-10896

### CODE REVIEW

File: backend/src/main/java/ai/verta/modeldb/experimentRun/FutureExperimentRunDAO.java [hunk 1]
```diff
@@ -838,4 +874,111 @@ public FutureExperimentRunDAO(
                 .build(),
         executor);
   }
+
+  /**
+   * @param expRunIds : ExperimentRun ids
+   * @return {@link Map<String, Map<String, CodeBlob >>} : Map from experimentRunID to Map of
+   *     LocationString to CodeVersion
+   */
+  private InternalFuture<Map<String, Map<String, CodeVersion>>> getExperimentRunCodeVersionMap(
```
Comment (conradoverta) [inline backend/src/main/java/ai/verta/modeldb/experimentRun/FutureExperimentRunDAO.java]: Can we move this to a helper like everything else? Let's keep the core DAO for orchestrating pieces. It helps avoid adding a ton of methods, which makes it hard to understand

File: backend/src/main/java/ai/verta/modeldb/experimentRun/FutureExperimentRunDAO.java [hunk 2]
```diff
@@ -699,7 +712,7 @@ public FutureExperimentRunDAO(
                                                     hyperparams.get(builder.getId()))),
                                     executor);
 
-                            final var futureHyperparamsFromConfigBlobs =
+                            var repoIdsFutureTasks =
```
Comment (conradoverta) [inline backend/src/main/java/ai/verta/modeldb/experimentRun/FutureExperimentRunDAO.java]:
```suggestion
                            final var repoIdsFutureTasks =
```

File: backend/src/main/java/ai/verta/modeldb/experimentRun/FutureExperimentRunDAO.java (hunk 1 above)
Comment (conradoverta) [inline backend/src/main/java/ai/verta/modeldb/experimentRun/FutureExperimentRunDAO.java]: Please add a note of where did you copy this from

Comment (anandjakhaniya) [reply backend/src/main/java/ai/verta/modeldb/experimentRun/FutureExperimentRunDAO.java]: Extracted on separate CodeVersionFromBlobHandler

File: backend/src/main/java/ai/verta/modeldb/experimentRun/FutureExperimentRunDAO.java (hunk 2 above)
Comment (anandjakhaniya) [reply backend/src/main/java/ai/verta/modeldb/experimentRun/FutureExperimentRunDAO.java]: Added

File: backend/src/main/java/ai/verta/modeldb/experimentRun/FutureExperimentRunDAO.java (hunk 1 above)
Comment (anandjakhaniya) [reply backend/src/main/java/ai/verta/modeldb/experimentRun/FutureExperimentRunDAO.java]: Copied from https://github.com/VertaAI/modeldb/blob/00f2554cf63f96d53ead230ea6340fed1a146120/backend/src/main/java/ai/verta/modeldb/experimentRun/ExperimentRunDAORdbImpl.java#L1690

File: backend/src/main/java/ai/verta/modeldb/experimentRun/subtypes/CodeVersionFromBlobHandler.java [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+        String versioningLocation = (String) objects[1];
+        GitCodeBlobEntity gitBlobEntity = (GitCodeBlobEntity) objects[2];
+        NotebookCodeBlobEntity notebookCodeBlobEntity = (NotebookCodeBlobEntity) objects[3];
+        PathDatasetComponentBlobEntity pathDatasetComponentBlobEntity =
+            (PathDatasetComponentBlobEntity) objects[4];
+
+        CodeVersion.Builder codeVersionBuilder = CodeVersion.newBuilder();
+        LOGGER.debug("notebookCodeBlobEntity {}", notebookCodeBlobEntity);
+        LOGGER.debug("pathDatasetComponentBlobEntity {}", pathDatasetComponentBlobEntity);
+        LOGGER.debug("gitBlobEntity {}", gitBlobEntity);
```
Comment (conradoverta) [inline backend/src/main/java/ai/verta/modeldb/experimentRun/subtypes/CodeVersionFromBlobHandler.java]: trace

File: backend/src/main/java/ai/verta/modeldb/experimentRun/FutureExperimentRunDAO.java [hunk 4]
```diff
@@ -751,6 +755,28 @@ public FutureExperimentRunDAO(
...
+                                            builder -> {
+                                              if (!runCodeVersionConfigBlob.isEmpty()
+                                                  && runCodeVersionConfigBlob.containsKey(
+                                                      builder.getId())) {
+                                                builder.putAllCodeVersionFromBlob(
+                                                    runCodeVersionConfigBlob.get(builder.getId()));
+                                              }
+                                              return builder;
+                                            }),
+                                    executor);
```
Comment (conradoverta) [inline backend/src/main/java/ai/verta/modeldb/experimentRun/FutureExperimentRunDAO.java]: The more I read this code pattern, the more I get a sense that it shouldn't be here polluting the main method but instead hidden in the handler. We can do as a follow up, but this method is _gigantic_ already and really hard for a reader to follow

File: backend/src/main/java/ai/verta/modeldb/experimentRun/FutureExperimentRunDAO.java [hunk 5]
```diff
@@ -719,23 +722,24 @@ public FutureExperimentRunDAO(
...
-                                                .thenApply(MapSubtypes::from, executor);
-                                          } else {
-                                            return hyperparametersFromConfigHandler
-                                                .getExperimentRunHyperparameterConfigBlobMap(
-                                                    new ArrayList<>(ids), selfAllowedRepositoryIds);
-                                          }
-                                        },
                                         executor);
+
+                            final var futureHyperparamsFromConfigBlobs =
```
Comment (conradoverta) [inline backend/src/main/java/ai/verta/modeldb/experimentRun/FutureExperimentRunDAO.java]: Comments

File: backend/src/main/java/ai/verta/modeldb/experimentRun/FutureExperimentRunDAO.java [hunk 6]
```diff
@@ -751,6 +755,28 @@ public FutureExperimentRunDAO(
                                             }),
                                     executor);
 
+                            final var futureCodeVersionFromBlob =
```
Comment (conradoverta) [inline backend/src/main/java/ai/verta/modeldb/experimentRun/FutureExperimentRunDAO.java]: Comments

**Review by conradoverta — APPROVED** [review]

**Review by conradoverta — APPROVED** [review]
Nevermind, just checked the commit itself

**Comment by anandjakhaniya** [comment]
Some logic was affected and need to change after other PRs merged so fix logic according to other PR review comment which is relevant to this.

**Pull request merged**